columnflow/tasks/framework/histograms.py

```diff
@@ -0,0 +1,154 @@
+# coding: utf-8
+
+"""
+Base tasks for working with multiple merged histograms.
+"""
+
+
+import law
+
+from columnflow.tasks.framework.base import Requirements, ShiftTask
+from columnflow.tasks.framework.mixins import (
+    CalibratorsMixin, SelectorStepsMixin, ProducersMixin, MLModelsMixin, WeightProducerMixin,
+    VariablesMixin, DatasetsProcessesMixin, CategoriesMixin,
+    ShiftSourcesMixin,
+)
+from columnflow.tasks.histograms import MergeHistograms, MergeShiftedHistograms
+from columnflow.util import dev_sandbox, maybe_import
+
+ak = maybe_import("awkward")
+
+
+class HistogramsUserBase(
+    DatasetsProcessesMixin,
+    CategoriesMixin,
+    VariablesMixin,
+    MLModelsMixin,
+    WeightProducerMixin,
+    ProducersMixin,
+    SelectorStepsMixin,
+    CalibratorsMixin,
+):
+    sandbox = dev_sandbox(law.config.get("analysis", "default_columnar_sandbox"))
+
+    def store_parts(self):
+        parts = super().store_parts()
+        parts.insert_before("version", "datasets", f"datasets_{self.datasets_repr}")
+        return parts
+
+    def load_histogram(self, dataset, variable):
+        histogram = self.input()[dataset]["collection"][0]["hists"].targets[variable].load(formatter="pickle")
+        return histogram
+
+    def slice_histogram(self, histogram, processes, categories, shifts, reduce_axes: bool = False):
```

Comment:
These are very useful function for a variaty of applications! It would be very nice if you could add doc strings and type hints in these two functions, since they are also intended to be used by users in the end